src/app/drive/components/FileViewer/utils/fileViewerWrapperUtils.ts

```diff
@@ -0,0 +1,222 @@
+        onMoveToTrashButtonClicked();
+        onClose();
+      },
+    });
+  };
+
+  const sharedActionsMenu = (): ListItemMenu<AdvancedSharedItem> => {
+    return contextMenuDriveItemSharedAFS({
+      openShareAccessSettings: isOwner ? onLinkSettingsButtonClicked : undefined,
+      copyLink: onCopyLinkButtonClicked,
+      deleteLink: () => undefined,
+      showDetails: onShowDetailsButtonClicked,
+      renameItem: !isCurrentUserViewer() ? onRenameItemButtonClicked : undefined,
+      moveItem: isOwner ? onMoveItemButtonClicked : undefined,
+      downloadItem: onDownloadItemButtonClicked,
+      moveToTrash: isOwner ? onShowStopSharingDialog : undefined,
+    });
+  };
+
+  const trashActionsMenu = (): ListItemMenu<DriveItemData> => {
+    return contextMenuTrashItems({
+      restoreItem: onRestoreItemButtonClicked,
+      deletePermanently: onDeletePermanentlyButtonClicked,
+    });
+  };
+
+  const contextMenuActions = () => {
+    if (isRecentsView) {
+      return recentsActionsMenu();
+    } else if (isSharedView) {
+      return sharedActionsMenu();
+    } else if (isTrashView) {
+      return trashActionsMenu();
+    } else {
+      return driveActionsMenu();
+    }
+  };
+
+  return contextMenuActions();
+};
+
+export function getFileContentManager(currentFile, downloadFile, handleFileThumbnail) {
+  const abortController = new AbortController();
+
+  return {
+    download: async (): Promise<Blob> => {
+      const shouldFileBeCached = canFileBeCached(currentFile);
+
+      const fileSource = await getDatabaseFileSourceData({ fileId: currentFile.id });
+      const isCached = !!fileSource;
+      let fileContent: Blob;
+
+      if (isCached) {
+        const isCacheExpired = !fileSource?.updatedAt
+          ? true
+          : dateService.isDateOneBefore({
+              dateOne: fileSource?.updatedAt,
+              dateTwo: currentFile?.updatedAt,
+            });
+        if (isCacheExpired) {
+          fileContent = await downloadFile(currentFile, abortController);
+          await updateDatabaseFileSourceData({
+            folderId: currentFile.folderId,
+            sourceBlob: fileContent,
+            fileId: currentFile.id,
+            updatedAt: currentFile.updatedAt,
+          });
+        } else {
+          fileContent = fileSource.source as Blob;
+          await handleFileThumbnail(currentFile, fileSource.source as File);
+        }
+      } else {
+        fileContent = await downloadFile(currentFile, abortController);
+        if (shouldFileBeCached) {
+          await updateDatabaseFileSourceData({
+            folderId: currentFile.folderId,
+            sourceBlob: fileContent,
+            fileId: currentFile.id,
+            updatedAt: currentFile.updatedAt,
+          });
+        }
+      }
+
+      return fileContent;
+    },
+    abort: () => {
+      abortController.abort();
+    },
+  };
+}
+
+export const useKeyboardShortcuts = ({
+  sharedKeyboardShortcuts,
+  driveItemActions,
```

Comment:
why is this coupled to driveItemActions? for me this looks like should be "product independent" as is not business logic